### ISSUE TITLE
Preview show error when change name of Cover Image field

### ISSUE THREAD

**Issue by manlm** [netlify/netlify-cms#5907]
[](url)I am using the default config.xml, when I change the name of the cover image field to "featureImage" like below:
` - label: 'Cover Image'
        name: 'featureImage'
        widget: 'image'
        required: false
        tagname: ''`

the preview feature show this error:
![Screen Shot 2021-10-17 at 22 53 46](https://user-images.githubusercontent.com/12764063/137635071-956cc07b-a67b-4546-b7b5-8d73612bacbb.png)

Is this a bug or I must have a field with the name "image".
Thank you.

**Comment by erezrokah** [comment]
Hi @manlm, that field is also referenced by a custom preview template:
https://github.com/netlify/netlify-cms/blob/9627452de0eb15201a06de93d9d03e22fe0be9e0/dev-test/index.html#L123
so when you rename it, the template breaks.

I'm keeping this issue open in case anyone would like to improve the error message when it happens.

**Comment by devnamrits** [comment]
Hi @erezrokah, what error message should be displayed when this happens?

Or I need to configure things in a way such that  if we do change anything here:
https://github.com/netlify/netlify-cms/blob/9627452de0eb15201a06de93d9d03e22fe0be9e0/dev-test/config.yml#L47-L51

if the name and the widget change then it should be changed in the following places as well:
https://github.com/netlify/netlify-cms/blob/9627452de0eb15201a06de93d9d03e22fe0be9e0/dev-test/index.html#L123
https://github.com/netlify/netlify-cms/blob/9627452de0eb15201a06de93d9d03e22fe0be9e0/dev-test/index.html#L185
Am I right?

What should I go for?

**Comment by erezrokah** [comment]
Hi @devnamrits, yes that's tricky as the error is coming from user custom code that we don't control, and can't be sure of the underlying issue.

How about showing a part of the stack trace, so users know where the error originated from?

**Comment by devnamrits** [comment]
Hi @erezrokah , Then I'm integrating something like that. Since we have a common ErrorBoundary.js file for producing all types of error in UI as show below:
https://github.com/netlify/netlify-cms/blob/58d67963036509f0426ef1351e28e25f5c3eb897/packages/netlify-cms-core/src/components/UI/ErrorBoundary.js#L142-L145
So, if we show some stack trace, then it will be applied to all other kinds UI error as well. So we're good to go with that, Right?

**Comment by erezrokah** [comment]
> So, if we show some stack trace, then it will be applied to all other kinds UI error as well. So we're good to go with that, Right?

Yes I think we're ok. Probably best to explicitly generate some errors to see how it looks in the UI

**Comment by devnamrits** [comment]
@erezrokah 
Yepp, I'll try to integrate that and try different scenarios by generating intentional errors and then will get back to you! Thanks for your kind advice.

**Comment by devnamrits** [comment]
<img width="1426" alt="Screenshot 2022-03-08 at 9 50 07 PM" src="https://user-images.githubusercontent.com/54183698/157281025-c4cceae5-6797-4b75-91ad-7775a4ebdff8.png">

Hi @erezrokah, Will this work fine?? I tried producing a few more errors. The error stack trace was there. If it's okay then I'll raise PR for it.

**Comment by devnamrits** [comment]
Can you please elaborate? I'm not getting exactly this PR and draft PR thing. Are you saying to me to deploy a netlify-cms app with GitHub backend and then raise an error and see how it looks? 

In the last comment, I was talking about just raising pull request for the changes that I made to display the stack trace.

**Comment by erezrokah** [comment]
> In the last comment, I was talking about just raising pull request for the changes that I made to display the stack trace.

Sorry for the confusion - when you open a PR, there's an option to open it as a draft PR, see https://github.blog/2019-02-14-introducing-draft-pull-requests/.

It communicates that it's a work in progress
